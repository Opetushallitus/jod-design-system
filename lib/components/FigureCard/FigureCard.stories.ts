import type { Meta, StoryObj } from '@storybook/react';

import { FigureCard } from './FigureCard';

const meta = {
  title: 'Primitives/Cards/FigureCard',
  component: FigureCard,
  tags: ['autodocs'],
} satisfies Meta<typeof FigureCard>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  parameters: {
    design: {
      type: 'figma',
      url: 'https://www.figma.com/file/LP7Acqn5dNNHS0R1YUWM1n/cx_jod_sb?node-id=1-1059',
    },
    docs: {
      description: {
        story: 'This is a card component for displaying a text figure with a caption.',
      },
    },
  },
  args: {
    content: 734,
    caption: 'ammatteja',
  },
};

export const Longer: Story = {
  parameters: {
    design: {
      type: 'figma',
      url: 'https://www.figma.com/file/LP7Acqn5dNNHS0R1YUWM1n/cx_jod_sb?node-id=1%3A1062',
    },
    docs: {
      description: {
        story: 'This is a card component for displaying a longer text figure with a caption.',
      },
    },
  },
  args: {
    content: 2336,
    caption: 'koulutuksia',
  },
};
