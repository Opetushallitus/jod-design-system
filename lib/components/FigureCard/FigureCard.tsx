export interface FigureCardProps {
  /** Content shown on the card */
  content: string | number;
  /** Caption shown on the card */
  caption: string;
}

/**
 * This is a card component for displaying a text figure with a caption.
 */
export const FigureCard = ({ content, caption }: FigureCardProps) => {
  return (
    <figure className="flex w-min flex-col-reverse items-center rounded-[10px] bg-jod-light px-[40px] py-[15px] font-bold">
      <figcaption className="text-[24px] leading-[26.4px] text-[#4D5358]">{caption}</figcaption>
      <span className="text-[80px] leading-[88px] text-jod-black">{content}</span>
    </figure>
  );
};
