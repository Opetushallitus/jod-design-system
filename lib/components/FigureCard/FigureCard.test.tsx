import { afterEach, describe, expect, it } from 'vitest';
import { render, screen, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/vitest';

import { FigureCard } from './FigureCard';

afterEach(() => {
  cleanup();
});

describe('FigureCard', () => {
  const content = 'Lorem ipsum';
  const caption = 'Dolor sit amet';

  it('renders the content and the caption', () => {
    render(<FigureCard content={content} caption={caption} />);
    expect(screen.getByText(content)).toBeInTheDocument();
    expect(screen.getByText(caption)).toBeInTheDocument();
    expect(screen.getByRole('figure')).toMatchSnapshot();
  });
});
