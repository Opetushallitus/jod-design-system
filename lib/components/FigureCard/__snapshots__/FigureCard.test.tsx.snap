// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FigureCard > renders the content and the caption 1`] = `
<figure
  class="flex w-min flex-col-reverse items-center rounded-[10px] bg-jod-light px-[40px] py-[15px] font-bold"
>
  <figcaption
    class="text-[24px] leading-[26.4px] text-[#4D5358]"
  >
    Dolor sit amet
  </figcaption>
  <span
    class="text-[80px] leading-[88px] text-jod-black"
  >
    Lorem ipsum
  </span>
</figure>
`;
